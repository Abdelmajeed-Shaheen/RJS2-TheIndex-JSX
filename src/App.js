import React from "react";
import authors from "./data";
import Container from "./Container"
import SideBar from "./SideBar"

function App() {
  return (
    <div id="app" className="container-fluid">
      <div className="row">
        <SideBar />
        <Container 
          authorsList={authors}
        />
      </div>
    </div>
  );
}


export default App;
