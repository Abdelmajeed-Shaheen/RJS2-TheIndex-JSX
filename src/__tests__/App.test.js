import React from 'react'
import { mount } from "enzyme";
import toJSON from "enzyme-to-json";

import App from "../App";

describe("<App />", () => {
  it("renders correctly", () => {
    const wrapper = mount(<App />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
