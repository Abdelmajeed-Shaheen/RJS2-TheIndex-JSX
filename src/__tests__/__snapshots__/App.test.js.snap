// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders correctly 1`] = `
<App>
  <div
    className="container-fluid"
    id="app"
  >
    <div
      className="row"
    >
      <div
        className="col-2"
      >
        <div
          id="sidebar"
        >
          <img
            alt="the index logo"
            className="logo"
            src="theindex.svg"
          />
          <section>
            <h4
              className="menu-item active"
            >
              <button>
                AUTHORS
              </button>
            </h4>
          </section>
        </div>
      </div>
      <div
        className="content col-10"
      >
        <div
          className="authors"
        >
          <h3>
            Authors
          </h3>
          <div
            className="row"
          >
            <div
              className="col-lg-4 col-md-6 col-12"
            >
              <div
                className="card"
              >
                <div
                  className="image"
                >
                  <img
                    alt="J.K. Rowling"
                    className="card-img-top img-fluid"
                    src="https://nation.com.pk/digital_images/large/2015-05-10/jk-rowling-believes-in-standing-up-to-bullies-1431261222-3596.jpeg"
                  />
                </div>
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    <span>
                      J.K. Rowling
                    </span>
                  </h5>
                  <small
                    className="card-text"
                  >
                    7
                     books
                  </small>
                </div>
              </div>
            </div>
            <div
              className="col-lg-4 col-md-6 col-12"
            >
              <div
                className="card"
              >
                <div
                  className="image"
                >
                  <img
                    alt="George R.R. Martin"
                    className="card-img-top img-fluid"
                    src="https://typeset-beta.imgix.net/elite-daily/2016/03/06144207/elite-daily-George-RR-Martin.jpg?w=800&h=800&auto=format&q=70&fit=crop&crop=faces"
                  />
                </div>
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    <span>
                      George R.R. Martin
                    </span>
                  </h5>
                  <small
                    className="card-text"
                  >
                    5
                     books
                  </small>
                </div>
              </div>
            </div>
            <div
              className="col-lg-4 col-md-6 col-12"
            >
              <div
                className="card"
              >
                <div
                  className="image"
                >
                  <img
                    alt="J.R.R Tolkein"
                    className="card-img-top img-fluid"
                    src="https://images.8tracks.com/cover/i/010/010/920/tolkeinbigger-2362.jpg?rect=166,0,500,500&q=98&fm=jpg&fit=max"
                  />
                </div>
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    <span>
                      J.R.R Tolkein
                    </span>
                  </h5>
                  <small
                    className="card-text"
                  >
                    5
                     books
                  </small>
                </div>
              </div>
            </div>
            <div
              className="col-lg-4 col-md-6 col-12"
            >
              <div
                className="card"
              >
                <div
                  className="image"
                >
                  <img
                    alt="Margaret Atwood"
                    className="card-img-top img-fluid"
                    src="https://literature.britishcouncil.org/assets/Uploads/writers/_resampled/CroppedImage680680-Margaret-Atwood-c-Jean-Malek-web.jpg"
                  />
                </div>
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    <span>
                      Margaret Atwood
                    </span>
                  </h5>
                  <small
                    className="card-text"
                  >
                    4
                     books
                  </small>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</App>
`;
